### ISSUE TITLE
Fine-Tuning the tensorflow model

### ISSUE THREAD

**Issue by arghavanmo** [tech-srl/code2vec#119]
Hi,
I have a question about the code2vec model which is trainable. Do you have any idea how I can fine-tune this model with a new dataset, with different purpose and labels. It is a binary classifier problem. Thus, the labels aren't methods name and these are class1 and class2.
I already checked your code but not sure in which file I should apply changes.

Thank you in advance for your help!

**Comment by urialon** [comment]
Hi @arghavanmo ,
Thank you for your interest in code2vec!

Yes, you can modify the code to fine-tune the model on a new dataset.
You will probably need to modify the target vocabulary to be a new trainable vocabulary of 2 label vectors, and also change the dictionary that maps from string labels to their indices.
I advise you to load the model and see where the target vocabulary is used, line by line, especially in these files:

https://github.com/tech-srl/code2vec/blob/master/model_base.py
https://github.com/tech-srl/code2vec/blob/master/tensorflow_model.py
https://github.com/tech-srl/code2vec/blob/master/vocabularies.py

Let me know if you have any further specific questions.
Best,
Uri
